### PR TITLE
backend/enh: moves whitelist to merchant level

### PR DESCRIPTION
**Pull request by jaypalmudaliyar24**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

1. Moves whitelisting check at merchant level.
2. Prepends `0` in mobile number of rider for OFFUS transactions.

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

1. Adds `merchant_id` column in `white_list_org` table.

## Motivation and Context

1. To enable whitelisting of NPs for particular merchants.
2. To give correct mobile number to call for OFFUS transactions

## How did you test it?

Tested locally.

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/dev/migrations/rider-app/1252-add-merchant-id-to-whitelistorg.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+WITH MerchantIds AS (
+    SELECT id as merchant_id FROM atlas_app.merchant
+)
+INSERT INTO atlas_app.white_list_org (id, subscriber_id, domain, merchant_id)
+SELECT
+    md5(random()::TEXT || clock_timestamp()::TEXT || random()::TEXT || random()::TEXT || random()::TEXT || random()::TEXT)::UUID,
+    subscriber_id,
+    domain,
+    MerchantIds.merchant_id as merchant_id
+FROM atlas_app.white_list_org CROSS JOIN MerchantIds;
```
Comment (NikithShetty) [inline Backend/dev/migrations/rider-app/1252-add-merchant-id-to-whitelistorg.sql:13]: Add comment which to run before and after migration

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/WhiteListOrg.hs [hunk 2]
```diff
@@ -30,8 +31,21 @@ import Kernel.Utils.Common
...
+findBySubscriberIdAndDomainAndMerchantId subscriberId domain merchantId = do
+  findOneWithKV
+    [ Se.Is BeamBLO.subscriberId $ Se.Eq $ getShortId subscriberId,
+      Se.Is BeamBLO.domain $ Se.Eq domain,
+      Se.Is BeamBLO.merchantId $ Se.Eq $ getId merchantId
+    ]
+
 findBySubscriberIdAndDomain :: (CacheFlow m r, EsqDBFlow m r) => ShortId Subscriber -> Domain -> m (Maybe WhiteListOrg)
-findBySubscriberIdAndDomain subscriberId domain = findOneWithKV [Se.Is BeamBLO.subscriberId $ Se.Eq $ getShortId subscriberId, Se.Is BeamBLO.domain $ Se.Eq domain]
+findBySubscriberIdAndDomain subscriberId domain = do
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/WhiteListOrg.hs:43]: Add comment for deprecating this

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/WhiteListOrg.hs (hunk 2 above)
Comment (jaypalmudaliyar24) [reply Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/WhiteListOrg.hs:44]: Fixed.

File: Backend/dev/migrations/rider-app/1252-add-merchant-id-to-whitelistorg.sql (hunk 1 above)
Comment (jaypalmudaliyar24) [reply Backend/dev/migrations/rider-app/1252-add-merchant-id-to-whitelistorg.sql:14]: Fixed.

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**